### PR TITLE
Reducing the erroneous download events to send only periodically

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
Also, activating the skipped unittests.

Issue # 

---

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/walinuxagent/1567)

### CODE REVIEW

File: azurelinuxagent/ga/exthandlers.py [hunk 1]
```diff
@@ -441,7 +441,17 @@ def handle_ext_handler(self, ext_handler, etag):
     def handle_handle_ext_handler_error(self, ext_handler_i, e, code=-1):
         msg = ustr(e)
         ext_handler_i.set_handler_status(message=msg, code=code)
-        ext_handler_i.report_event(message=msg, is_success=False, log_event=True)
+
+        if isinstance(e, ExtensionDownloadError):
+            self.get_artifact_error_state.incr()
+            if self.get_artifact_error_state.is_triggered():
```
Comment (narrieta) [inline azurelinuxagent/ga/exthandlers.py]: ok to send the event periodically, but the error should still be logged, right?

File: azurelinuxagent/ga/exthandlers.py [hunk 2]
```diff
@@ -623,7 +633,7 @@ def decide_version(self, target_state=None):
         try:
             pkg_list = self.protocol.get_ext_handler_pkgs(self.ext_handler)
         except ProtocolError as e:
-            raise ExtensionError("Failed to get ext handler pkgs", e)
+            raise ExtensionDownloadError("Failed to get ext handler pkgs", e)
```
Comment (narrieta) [inline azurelinuxagent/ga/exthandlers.py]: get_ext_handler_pkgs() may raise ProtocolError for reasons other than download errors. I think you want to make the conversion to DownloadError deeper in the code, probably somewhere in the call to or implementation of get_ext_manifest

File: azurelinuxagent/ga/exthandlers.py (hunk 1 above)
Comment (vrdmr) [reply azurelinuxagent/ga/exthandlers.py]: For `report_event` function, the default value for param `log_event` is True. Which in this case (as `is_success` is False) would log the line as well. I'll make it explicit.

File: azurelinuxagent/ga/exthandlers.py (hunk 2 above)
Comment (vrdmr) [reply azurelinuxagent/ga/exthandlers.py]: Thanks for the tip. Went through the call tree. Will take another look to narrow down the exception causing areas.

**Pull request merged**